driver/mloKthValueHost.hpp

```diff
@@ -0,0 +1,125 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_view_utils.hpp>
+
+#include <../test/ford.hpp>
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloKthvalueFwdRunHost(const miopenTensorDescriptor_t inputDesc,
+                              const miopenTensorDescriptor_t outputDesc,
+                              const miopenTensorDescriptor_t indicesDesc,
+                              const Tgpu* input,
+                              Tcheck* output,
+                              size_t* indices,
+                              const size_t k,
+                              const uint64_t dim)
+{
+    size_t input_size = miopen::deref(inputDesc).GetElementSize();
+    size_t dim_size   = miopen::deref(inputDesc).GetLengths()[dim];
+    size_t dim_stride = miopen::deref(inputDesc).GetStrides()[dim];
+
+    auto input_tv             = miopen::get_inner_expanded_tv<5>(miopen::deref(inputDesc));
+    auto input_tv_without_dim = miopen::get_tv_without_dim<5>(input_tv, dim);
+    auto output_tv            = miopen::get_inner_expanded_tv<5>(miopen::deref(outputDesc));
+    auto indices_tv           = miopen::get_inner_expanded_tv<5>(miopen::deref(indicesDesc));
+
+    size_t num_slice = input_size / dim_size;
+
+    std::vector<float> elements;
+    std::vector<size_t> ids(dim_size);
+    for(size_t i = 0; i < dim_size; ++i)
+    {
+        ids[i] = i;
+    }
+
+    for(size_t slide_id = 0; slide_id < num_slice; ++slide_id)
+    {
+        elements.clear();
+        tensor_layout_t<4> layout(input_tv_without_dim, slide_id);
+        auto idx = input_tv_without_dim.get_tensor_view_idx(layout);
+
+        for(int j = 0; j < dim_size; ++j)
+        {
+            elements.push_back(static_cast<float>(input[idx + j * dim_stride]));
+        }
+
+        std::sort(ids.begin(), ids.end(), [=](size_t x, size_t y) -> bool {
+            return elements[x] < elements[y];
+        });
+
+        auto output_layout  = tensor_layout_t<5>(output_tv, slide_id);
+        auto indices_layout = tensor_layout_t<5>(indices_tv, slide_id);
+        output[output_tv.get_tensor_view_idx(output_layout)] =
+            static_cast<Tcheck>(elements[ids[k - 1]]);
+        indices[indices_tv.get_tensor_view_idx(indices_layout)] = ids[k - 1];
+    }
```

Comment:
`par_ford` can be used, right?

Reply:
Apply it for `cpu_kthvalue.hpp` too.

Reply:
yes, you can use par_ford here, i'll fix it

Reply:
solved